app/src/main/java/org/simple/clinic/consent/onboarding/OnboardingConsentScreenFragment.kt

```diff
@@ -0,0 +1,208 @@
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawWithCache
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.fragment.app.Fragment
+import kotlinx.parcelize.Parcelize
+import org.simple.clinic.R
+import org.simple.clinic.common.ui.components.FilledButton
+import org.simple.clinic.common.ui.theme.SimpleTheme
+import org.simple.clinic.consent.onboarding.OnboardingConsentEvent.AgreeButtonClicked
+import org.simple.clinic.di.injector
+import org.simple.clinic.navigation.v2.ScreenKey
+import org.simple.clinic.registerorlogin.AuthenticationActivity
+import org.simple.clinic.util.disableAnimations
+import org.simple.clinic.util.finishWithoutAnimations
+import org.simple.clinic.util.mobiusViewModels
+import org.simple.clinic.util.unsafeLazy
+import javax.inject.Inject
+
+class OnboardingConsentScreenFragment : Fragment(), UiActions {
+
+  @Inject
+  lateinit var effectHandlerFactory: OnboardingConsentEffectHandler.Factory
+
+  private val viewEffectHandler by unsafeLazy { OnboardingConsentViewEffectHandler(this) }
+  private val viewModel by mobiusViewModels(
+      defaultModel = { OnboardingConsentModel },
+      update = { OnboardingConsentUpdate() },
+      effectHandler = { viewEffectsConsumer -> effectHandlerFactory.create(viewEffectsConsumer).build() }
+  )
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    context.injector<Injector>().inject(this)
+  }
+
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+    return ComposeView(requireContext()).apply {
+      setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+      setContent {
+        SimpleTheme {
+          OnboardingConsentScreen(
+              onAccept = { viewModel.dispatchEvent(AgreeButtonClicked) }
+          )
+        }
+      }
+    }
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    viewModel.viewEffects.setObserver(
+        viewLifecycleOwner,
+        { viewEffect -> viewEffectHandler.handle(viewEffect) },
+        { pausedViewEffects -> pausedViewEffects.forEach { viewEffectHandler.handle(it) } }
+    )
+  }
+
+  override fun moveToRegistrationActivity() {
+    // This navigation should not be done here, we need a way to publish
+    // an event to the parent activity (maybe via the screen router's
+    // event bus?) and handle the navigation there.
+    // TODO(vs): 2019-11-07 Move this to an event that is subscribed in the parent activity
```

Comment:
Are we still going to improve `moveToRegistrationActivity`? Can we remove the comment from here and convert it into shortcut story?

Reply:
Yes, eventually when we get back to working on Simple. We can create a new story, but let's leave the TODO in the project. IDE provides a good way to track all the TODOs.